test/models/report_test.rb

```diff
@@ -3,7 +3,32 @@
 require 'test_helper'
 
 class ReportTest < ActiveSupport::TestCase
-  # test "the truth" do
-  #   assert true
-  # end
+  fixtures :users, :reports
+
+  test '#editable?' do
+    report = reports(:alice_report)
+    target_user = report.user
+    assert report.editable?(target_user)
+  end
+
+  test '#created_on' do
+    report = reports(:alice_report)
+    assert_equal report.created_at.to_date, report.created_on
+  end
+
+  test '#save_mentions' do
+    alice = users(:alice)
+    alice_report = alice.reports.create!(title: 'alice_report', content: '自分が聴いだのは近頃途中でおもにないうでし。')
+
+    bob = users(:bob)
+    bob_report = bob.reports.create!(title: 'bob_report', content: "昨日こちらの日報を読みました。http://localhost:3000/reports/#{alice_report.id}")
+
+    assert bob_report.mentioning_reports.find(alice_report.id)
```

Comment:
findはデータが見つからないと例外を出すのでfalseの戻り値が返ることはないですね。
なので assert でテストするのには向いてないです。

31-32行目もそうですが、1つずつfindで検証していくのは大変なので、配列を比較するようにした方が良いと思います。

```suggestion
    assert_equal [alice_report], bob_report.mentioning_reports
```

Reply:
配列を使用する形に修正しました! 日報更新後のアサーションでは、以下QAを参考にキャッシュを破棄して再読み込みするためreloadを使用しています。
https://bootcamp.fjord.jp/questions/1821